### PR TITLE
Tutorial onnx part 2: Fine-tuning an ONNX model on Gluon

### PR DESCRIPTION
**Pull request by ThomasDelteil**
Site version testable here:
http://8c859074f1c7.ant.amazon.com:8080/tutorials/onnx/fine_tuning_onnx_gluon.html

@thomelane @saffrooze @Ishitori

### CODE REVIEW

File: docs/tutorials/onnx/fine_tuning_gluon.md [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+new_sym, new_params = get_layer_output(sym, params, 'relu6')
+```
+
+### Fine-tuning in gluon
+
+We pick a context
+
+
+```python
+ctx = mx.gpu()
```
Comment (Ishitori) [inline docs/tutorials/onnx/fine_tuning_gluon.md]: It will fail if a user doesn't have GPU (like running on mac). For safety reason I recommend to replace it with:
ctx = mx.gpu() if mx.test_utils.list_gpus() else mx.cpu()

File: docs/tutorials/onnx/fine_tuning_gluon.md [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+for epoch in range(20):
+    for i, (data, label) in enumerate(dataloader_train):
+        data = data.astype(np.float32).as_in_context(ctx)
+        label = label.as_in_context(ctx)
+        with autograd.record():
+            output = net(data)
+            loss = softmax_cross_entropy(output, label)
+        loss.backward()
+        trainer.step(data.shape[0])
+    nd.waitall() # wait at the end of the epoch
```
Comment (Ishitori) [inline docs/tutorials/onnx/fine_tuning_gluon.md]: For performance reason, it is better to put this line right after label = label.as_in_context(ctx), so if a user has GPU loading of the data and calculation happens simultaneously

File: docs/tutorials/onnx/fine_tuning_gluon.md [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+    new_val_accuracy = evaluate_accuracy_gluon(dataloader_test, net)
+    print("Epoch {}, Test Accuracy {} ".format(epoch, new_val_accuracy))
+    
+    if val_accuracy > new_val_accuracy:
```
Comment (Ishitori) [inline docs/tutorials/onnx/fine_tuning_gluon.md]: I would add an extra comment, that you are doing early stop regularization

File: docs/tutorials/onnx/fine_tuning_gluon.md (hunk 1 above)
Comment (Ishitori) [reply docs/tutorials/onnx/fine_tuning_gluon.md]: ...And warn a user that using CPU it would take forever to fine tune the model

Comment (ThomasDelteil) [reply docs/tutorials/onnx/fine_tuning_gluon.md]: Thanks sounds good, nice little one-liner 👍

File: docs/tutorials/onnx/fine_tuning_gluon.md (hunk 2 above)
Comment (ThomasDelteil) [reply docs/tutorials/onnx/fine_tuning_gluon.md]: This line is done at the end of the epoch, not at the end of the batch (so the entire dataset is loaded in memory at the beginning of the epoch already)

File: docs/tutorials/onnx/fine_tuning_gluon.md (hunk 3 above)
Comment (ThomasDelteil) [reply docs/tutorials/onnx/fine_tuning_gluon.md]: Thanks, will add the comments
